src/components/PopupWithForm.js

```diff
@@ -1,33 +1,40 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, submitButton) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupElement.querySelector(".modal__submit-button");
+    // this._originalButtonText = this._submitButton.textContent;
   }
 
   _getInputValues() {
-    //collects data from all the input fields
-    this._inputList = this._popupElement.querySelectorAll(".modal__input");
-    this._values = {};
-    this._inputList.forEach((input) => {
-      this._values[input.name] = input.value;
-    });
-
-    return this._values;
+    const inputList = this._popupElement.querySelectorAll(".modal__input");
+    const values = {};
+    inputList.forEach(input => values[input.name] = input.value);
+    return values;
   }
 
   setEventListeners() {
-    this._popupForm.addEventListener("submit", () => {
-      this._handleFormSubmit(this._getInputValues());
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      const formData = this._getInputValues();
+      this.setLoading(this._submitButton, true);
+      this._handleFormSubmit(formData); // Pass form values, not evt
       this.close();
     });
     super.setEventListeners();
   }
 
-  close() {
-    this._popupForm.reset();
-    super.close();
+  setLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = 'Saving...';
+    } else {
+      this._submitButton.textContent = 'Submit';
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```